### PR TITLE
LORIS crawlers used to crawl several of the CONP datasets

### PR DESCRIPTION
**Pull request by cmadjar**
This pulls the pipelines used to generate several of the CONP datasets hosted in LORIS:

- `datalad_crawler/pipelines/loris.py` pipeline was used to crawl:
  - [CONP preventad-open](https://github.com/conpdatasets/preventad-open) 
  - [CONP preventad-registered-images-in-MINC](https://github.com/conpdatasets/preventad-registered-images-in-MINC)
  - [CONP multicenter-phantom](https://github.com/conpdatasets/multicenter-phantom)
- `datalad_crawler/pipelines/loris_bids_export.py` pipeline was used to crawl:
  - [CONP preventad-open-bids](https://github.com/conpdatasets/preventad-open-bids) 
  - [CONP preventad-registered-images-in-BIDS](https://github.com/conpdatasets/preventad-registered-images-in-BIDS) 
- `datalad_crawler/pipelines/loris_data_releases.py` pipeline was used to crawl:  
  - [CONP preventad-registered-clinical-data](https://github.com/conpdatasets/preventad-registered-clinical-data)

### CODE REVIEW

**Comment by cmadjar** [comment]
Oooops. Very sorry, I meant to open this PR to the CONP-PCNO fork of datalad-crawler...

I was thinking of sending you an email to ask you if you would be interested in adding those crawlers to your code. Well, now you know my evil plan, haha ;).

Anyway, let me know if you would be interested. If not, no worries, we'll keep those in the CONP-PCNO fork.

Thank you!

File: datalad_crawler/pipelines/loris_bids_export.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+from logging import getLogger
+lgr = getLogger("datalad.crawler.pipelines.kaggle")
+
+
+class LorisAPIBIDSExtractor(object):
+
+    def __init__(self, apibase=None, annex=None):
+        self.apibase = apibase
+        self.meta    = {}
+        self.repo    = annex.repo
```
Comment (yarikoptic) [inline datalad_crawler/pipelines/loris_bids_export.py:31]: with the common code for `finalize`, constructor and majority of the pipeline, I think you could have avoided good amount of code duplication by establishing some `BaseLorisExtractor` to reside e.g. in `loris.py` and serve as a base class.  Then derived classes would provide only the critical difference (configuration for annex, these days also better be done via config procedures, but I guess ok as is for here/antique-crawler).

**Review by yarikoptic — COMMENTED** [review]
Thanks for sharing! Great to hear that crawler serves you (well?).  In general, I would not mind having this PR merged to enrich collection of pipelines -- might be easier for others to find later on to craft something similar.
But I wonder if you would consider improving upon it?

BTW -- is there some "public" loris instance on which at least some of those pipelines could be tested on some tiny dataset?

**Comment by yarikoptic** [comment]
seeing the code and `kaggle` logger being used -- is this a work incorporating earlier #67? (sorry I have missed it originally)

File: datalad_crawler/pipelines/loris_bids_export.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+                statusdb='json',
+                skip_problematic=True,
+                special_remotes=[DATALAD_SPECIAL_REMOTE],
+                options=[
+                    "-c",
+                    "annex.largefiles="
+                    "exclude=README.md and exclude=DATS.json and exclude=logo.png"
+                    " and exclude=.datalad/providers/loris.cfg"
+                    " and exclude=.datalad/crawl/crawl.cfg"
+                    " and exclude=*scans.json"
```
[the remark this answers is not in the document]
Comment (cmadjar) [reply datalad_crawler/pipelines/loris_bids_export.py:121]: these are the description of the `scans.tsv` so in theory, there is no data. I had to track the `scans.json` files in git as they all had the same hash and it led to tons of URLs for a given given and slowed down the download process of those tiny files. (Related to https://github.com/datalad/datalad/issues/5429)

File: datalad_crawler/pipelines/loris_bids_export.py (hunk 1 above)
Comment (cmadjar) [reply datalad_crawler/pipelines/loris_bids_export.py:31]: very true! I did them one at a time when I needed them and with not much time to do proper programming ;). I'll look into it.

**Comment by cmadjar** [comment]
@yarikoptic Thank you! I am definitely happy to improve the code :-).

The code here is indeed based on #13. I did not see the #67 PR. I was told the code #13 was used for PREVENT-AD so I reused it for the other LORIS instances of CONP datasets. I will check with @mathdugre to see the difference between the crawlers.

The datasets listed in the descriptions are all open (except the PREVENT-AD registered ones which is open only to PIs). However, none of the datasets are small unfortunately... Were you thinking of manual testing or automated testing? I could ask around to see what is possible.

Thank you!

File: datalad_crawler/pipelines/loris_bids_export.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (yarikoptic) [reply datalad_crawler/pipelines/loris_bids_export.py:121]: ha ha -- I mixed them up [**again**](https://github.com/datalad/datalad/issues/5429#issuecomment-783482145) -- indeed it is a good fit for plain git.

If you are involved with producing those BIDS datasets, there could be a singular `scans.json` on top level which should then be inherited for each subject/session, and thus avoiding need to duplicate them.  Unfortunately bids-specification is not explicitly mentioning that, so I filed https://github.com/bids-standard/bids-specification/issues/789 to possibly improve upon that.

**Comment by yarikoptic** [comment]
> Were you thinking of manual testing or automated testing? I could ask around to see what is possible.

automated would be the ultimate goal.  If no suitable smallish dataset is out there, I guess there could be some `include` or `exclude` option to limit to some subset of files to make test run succinct.

**Comment by cmadjar** [comment]
Just a little note to tell you to discard my last comment. I figured it out :).

Ultimately, I think all LORIS PRs will be closed and I will send a new one with improved crawlers. Might just take some time though. Will definitely keep you posted.

**Comment by yarikoptic** [comment]
My slowness was rewarded! ;-) no rush on my end, but I might come handy to review earlier version of RF

**Comment by cmadjar** [comment]
To be continued on #103

**Pull request closed without merging**